src/main/java/net/pms/dlna/protocolinfo/ProtocolInfoAttribute.java

```diff
@@ -0,0 +1,454 @@
+		 *            is only used when retrieving {@link DLNAOrgOperations}
+		 *            instances and can be {@code null} for other attribute
+		 *            types.
+		 * @return The existing instance or {@code null}.
+		 * @throws ParseException If {@code attributeValue} can't be parsed.
+		 */
+		public ProtocolInfoAttribute getAttribute(
+			String attributeName,
+			String attributeValue,
+			Protocol protocol
+		) throws ParseException {
+			return getAttribute(
+				ProtocolInfoAttributeName.FACTORY.createAttributeName(attributeName),
+				attributeValue,
+				protocol
+			);
+		}
+
+		/**
+		 * Retrieves a predefined or cached {@link ProtocolInfoAttribute}
+		 * instance if it exists. If no matching instance is found, {@code null}
+		 * is returned.
+		 * <p>
+		 * <b>Note:</b> {@link DLNAOrgFlags} isn't predefined or cached and
+		 * will always return {@code null}.
+		 * <p>
+		 * This method can only retrieve values in their {@link String} form.
+		 * Use the factory methods of the individual classes to retrieve
+		 * instances for values of other types.
+		 *
+		 * @param attributeName the {@link ProtocolInfoAttributeName}.
+		 * @param attributeValue the attribute {@link String} value.
+		 * @param protocol the {@link Protocol} of the {@code protocolInfo} this
+		 *            {@link ProtocolInfoAttribute} belongs to. This parameter
+		 *            is only used when retrieving {@link DLNAOrgOperations}
+		 *            instances and can be {@code null} for other attribute
+		 *            types.
+		 * @return The existing instance or {@code null}.
+		 * @throws ParseException If {@code attributeValue} can't be parsed.
+		 */
+		public ProtocolInfoAttribute getAttribute(
+			ProtocolInfoAttributeName attributeName,
+			String attributeValue,
+			Protocol protocol
+		) throws ParseException {
+			if (attributeName == null) {
+				return null;
+			}
+
+			/* Check predefined and cached types, the following classes don't have
+			 * predefined or cached types and will always return null:
+			 *
+			 * - DLNAOrgFlags
+			 *
+			 */
+			if (DLNAOrgConversionIndicator.NAME.equals(attributeName)) {
+				return DLNAOrgConversionIndicator.FACTORY.getConversionIndicator(attributeValue);
+			} else if (DLNAOrgOperations.NAME.equals(attributeName)) {
+				return DLNAOrgOperations.FACTORY.getOperations(protocol, attributeValue);
+			} else if (DLNAOrgPlaySpeeds.NAME.equals(attributeName)) {
+				return DLNAOrgPlaySpeeds.FACTORY.getPlaySpeeds(attributeValue);
+			} else if (DLNAOrgProfileName.NAME.equals(attributeName)) {
+				return DLNAOrgProfileName.FACTORY.getProfileName(attributeValue);
+			} else if (PanasonicComProfileName.NAME.equals(attributeName)) {
+				return PanasonicComProfileName.FACTORY.getProfileName(attributeValue);
+			} else if (AribOrJpProfileName.NAME.equals(attributeName)) {
+				return AribOrJpProfileName.FACTORY.getProfileName(attributeValue);
+			} else if (attributeName.getName().contains("_PN")) {
```

Comment:
1) } else if (attributeName.getName().contains("_PN")) {
2) } else if (DLNAOrgProfileName.NAME.equals(attributeName)

I'm too sleepy to even understand you answer.